### PR TITLE
Add tracing support for the quartus backend

### PR DESCRIPTION
**Pull request by jmitrevs**
# Description

Tracing was only partially supported in the Quartus backend. This adds the missing features to enable support in the currently supported `io_parallel` style. (This will need to be expanded when streaming is added.)

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Tests

`test_trace.py` is added in pytest:  it is a simple copy of the `test_dense` with tracing enabled.

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/master/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation -- not needed; should be identical as Vivado,
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.B# Description

### CODE REVIEW

**Comment by jmitrevs** [comment]
I just made pr/583 locally with this pull request.

**Comment by jmitrevs** [comment]
I think this is fairly straightforward and uncontroversial--mostly a copy of the Vivado behavior, updated for ac_fixed.  I would recommend reviewing it so we can merge it.

**Review by bo3z — CHANGES_REQUESTED** [review]
Looks good to me, happy to merge after removing the double declaration inside nnet_helpers.h

File: hls4ml/templates/quartus/firmware/nnet_utils/nnet_helpers.h [hunk 1]
```diff
@@ -58,6 +60,49 @@ constexpr int pow2(int x){
   return x == 0 ? 1 : 2 * pow2(x - 1);
 }
 
+extern bool trace_enabled;
+extern std::map<std::string, void *> *trace_outputs;
+extern size_t trace_type_size;
```
[the remark this answers is not in the document]
Comment (jmitrevs) [reply hls4ml/templates/quartus/firmware/nnet_utils/nnet_helpers.h:65]: You are right. I will remove them.

**Comment by bo3z** [comment]
All good, happy to merge.

**Pull request merged**